### PR TITLE
bump dependencies and configs to use webpack 5

### PR DESCRIPTION
**Pull request by solomontang**
- update to use webpack 5 and webpack-node-externals
- before externals
image
- after externals
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,100 +30,127 @@
...
+    "connected-react-router": "~6.9.0",
+    "core-js": "~3.12.0",
     "enzyme": "~3.11.0",
-    "enzyme-adapter-react-16": "~1.15.5",
-    "eslint": "~7.16.0",
+    "enzyme-adapter-react-16": "~1.15.6",
+    "eslint": "~7.27.0",
     "eslint-config-airbnb": "~18.2.1",
     "eslint-import-resolver-webpack": "~0.13.0",
     "eslint-loader": "~4.0.0",
```
Comment (UnsungHero97) [inline package.json:75]: You can remove this dependency, it's not used anywhere.

File: package.json [hunk 2]
```diff
@@ -30,100 +30,127 @@
...
+    "@fortawesome/react-fontawesome": "^0.1.0",
+    "@redux-saga/core": "^1.1.0",
+    "connected-react-router": "^6.9.0",
+    "history": "^4.10.0",
+    "immutable": "4.0.0-rc.10",
+    "lattice": "^0.92.0",
+    "lattice-auth": "~0.23.0",
+    "lattice-sagas": "^0.66.0",
+    "lattice-ui-kit": "~0.43.0",
+    "lattice-utils": "~0.11.0",
```
Comment (UnsungHero97) [inline package.json:138]: Why do you use both `~` and `^` here?

File: package.json [hunk 3]
```diff
@@ -30,100 +30,127 @@
...
+    "lodash": "^4.17.21",
+    "luxon": "^1.26.0",
+    "react": "^16.14.0",
+    "react-dom": "^16.14.0",
+    "react-redux": "^7.2.0",
+    "react-router": "^5.2.0",
+    "react-router-dom": "^5.2.0",
+    "redux": "^4.0.0",
+    "redux-immutable": "4.0.0",
+    "redux-reqseq": "~0.6.0",
```
Comment (UnsungHero97) [inline package.json:148]: Should this be `~` or `^`?

File: src/containers/profile/src/components/SurveyHistory.js [hunk 4]
```diff
@@ -7,9 +7,13 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { List as iList } from 'immutable';
 import { Constants } from 'lattice';
 import {
+  // $FlowFixMe
```
Comment (UnsungHero97) [inline src/containers/profile/src/components/SurveyHistory.js:10]: I think you can avoid these if you update `.flowconfig` with this:

```
[untyped]
<PROJECT_ROOT>/node_modules/lattice-ui-kit
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: config/jest/babelJestTransformer.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
 const path = require('path');
 
-module.exports = require('babel-jest').createTransformer({
+module.exports = require('babel-jest').default.createTransformer({
```
[the remark this answers is not in the document]
Comment (solomontang) [reply config/jest/babelJestTransformer.js:5]: yes

File: config/webpack/webpack.config.js [hunk 6]
```diff
@@ -1,11 +1,20 @@
-/* eslint-disable import/extensions */
+/* eslint-disable import/extensions, import/no-extraneous-dependencies */
 
 const path = require('path');
 const Webpack = require('webpack');
+const externals = require('webpack-node-externals');
+const TerserPlugin = require('terser-webpack-plugin');
 
-const LIB_CONFIG = require('../lib/lib.config.js');
-const LIB_PATHS = require('../lib/paths.config.js');
```
[the remark this answers is not in the document]
Comment (solomontang) [reply config/webpack/webpack.config.js:7]: true. these can be removed

File: config/babel/babel.config.js [hunk 7]
```diff
@@ -1,35 +1,46 @@
...
     }],
+    '@babel/plugin-proposal-class-properties',
+    'babel-plugin-styled-components',
   ],
   presets: [
-    '@babel/preset-env',
+    ['@babel/preset-env', {
+      corejs: 3,
+      useBuiltIns: 'entry',
+    }],
```
[the remark this answers is not in the document]
Comment (solomontang) [reply config/babel/babel.config.js]: updated. useBuiltIns is false by default so just gonna omit the options

**Review by UnsungHero97 — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Pull request merged**